services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -627,10 +627,16 @@ private void sendAdminMessageAndWaitForConsumed(String clusterName, String store
             adminCommandExecutionTracker.createExecution(AdminMessageType.valueOf(message).name());
         message.executionId = execution.getExecutionId();
         VeniceWriter<byte[], byte[], byte[]> veniceWriter = veniceWriterMap.get(clusterName);
-        byte[] serializedValue = adminOperationSerializer.serialize(message);
+        Map<String, Long> metadata = adminTopicMetadataAccessor.getMetadata(clusterName);
```

Comment:
In practice, all the Controllers are being shared by all the clusters, so when the Controllers are fully deployed, all the clusters will receive the upgrade at the same time.
It is fine to maintain the writer schema id per cluster, but it would cause some maintenance overhead and potentially, some misleading when adding new clusters.
I wondered whether we can use a separate Znode, which is outside of any cluster instead or not, and with this way, when updating the writer schema id via tooling, we don't need to update it cluster by cluster.

Reply:
Also shall we embed this writer schema retrieval logic into `AdminOperationSerializer`? so that the caller doesn't need to bother about how to fetch the correct writer schema id, which will keep `serialize` interface same as before and internally, it will figure out which writer schema is to use.
Not a big concern, and mainly for better encapsulation and reduce the complexity of `VeniceParentHelixAdmin`, which is already lengthy today.

Reply:
> shall we embed this writer schema retrieval logic into AdminOperationSerializer
I need to send the version along with the message itself ([code](https://github.com/linkedin/venice/pull/1549/files#diff-ed8e93ee89eeb6383b0604c2d5cc0bb816bce922d0e66cd593a51b88e62b67b2R639)), so I don't think we can put the get inside AdminOperationSerializer.

Reply:
> I wondered whether we can use a separate Znode, which is outside of any cluster instead or not, and with this way, when updating the writer schema id via tooling, we don't need to update it cluster by cluster.

This is a valid suggestion. I don't think we will cover that change in this PR. We can discuss more on this offline. Since the ZK needs to sit somewhere, I am not really clear whether we can get it out of "cluster" concept. Do we have any existing usage for the znode you are talking about?

Reply:
Regarding ZNode, we can certainly can create a new Znode outside of any cluster and currently `storeConfigs` are outside of any cluster since the info stored is global and I think the admin operation schema id is also global info and we can create a new znode in the same layer as `storeConfigs`.

Reply:
> shall we embed this writer schema retrieval logic into AdminOperationSerializer
> 
> I need to send the version along with the message itself (code), so I don't think we can put the get inside AdminOperationSerializer.
We can let `serialize` function return a pair to keep both bytes and writer schema id.
It is not a big concern, and it is up to you.
The main goal is to reduce the complexity of `VeniceParentHelixAdmin` and make the code more modular.

Reply:
Thanks @gaojieliu , let me discuss more with @huangminchn on this and get back to you offline. I will send out follow-up PR if we decide to move things. 

cc: @huangminchn for the option for znode location.

Reply:
> > shall we embed this writer schema retrieval logic into AdminOperationSerializer
> > I need to send the version along with the message itself (code), so I don't think we can put the get inside AdminOperationSerializer.
> 
> We can let `serialize` function return a pair to keep both bytes and writer schema id. It is not a big concern, and it is up to you. The main goal is to reduce the complexity of `VeniceParentHelixAdmin` and make the code more modular.

Discussed offline. I decided to keep it as it is since other options do not appear strongly better than the current option.

Reply:
What about the global write schema instead of per-cluster?
Have you got a chance to talk to @huangminchn regarding ^?

Reply:
We synced up offline on this topic, and decided to keep per cluster config to reduce blast radius. However, tooling will be provided to update all clusters at the same time.